### PR TITLE
Add fr.nytuo.cosmiccomics

### PR DESCRIPTION
**Pull request by Nytuo**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
        -> I pushed all the files needed inspirated from other electron apps (like ytmdesktop or slack)
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. 
        -> The app needs access to the full storage, internet connection and it's better to have a GPU accelerated devices since this is an electron app (with chromium)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
        -> The app follow the flatpak guide for building electron application.
- [x] I am an upstream contributor to the project. https://github.com/Nytuo/CosmicComics
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting. (https://nytuo.fr) The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: flathub.json
Comment (bbhtt) [inline flathub.json:1]: Delete, these are the only arches available at the moment

File: fr.nytuo.cosmiccomics.desktop
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.desktop:1]: Desktop file should be upstream

File: fr.nytuo.cosmiccomics.metainfo.xml
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.metainfo.xml:1]: Appdata file should be upstream

File: fr.nytuo.cosmiccomics.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+<component type="desktop-application">
+  <id>fr.nytuo.cosmiccomics</id>
+  
+  <name>Cosmic Comics</name>
+  <summary>Cosmic Comics is a Comics and Mangas reader and collectionner.</summary>
+  
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GNU GPLv3</project_license>
+
+  <content_rating type="oars-1.0" />
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.metainfo.xml:11]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: fr.nytuo.cosmiccomics.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>fr.nytuo.cosmiccomics</id>
+  
+  <name>Cosmic Comics</name>
+  <summary>Cosmic Comics is a Comics and Mangas reader and collectionner.</summary>
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.metainfo.xml:6]:
```suggestion
  <summary>Cosmic Comics is a Comics and Mangas reader and collectionner</summary>
```

File: fr.nytuo.cosmiccomics.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=host
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.yml:16]: This is pretty broad. Why do you need it? Electron supports using the filechooser portal.

File: fr.nytuo.cosmiccomics.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+build-options:
+  append-path: /usr/lib/sdk/node18/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: cosmiccomics
+    buildsystem: simple
+    build-options:
+      build-args:
+        - --share=network
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.yml:27]:
```suggestion
```

This has no effect on flathub. Builds will be done offline

File: fr.nytuo.cosmiccomics.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ./run.sh
+    sources:
+      #- type: archive
+      #  url: https://github.com/flathub/electron-sample-app/archive/1.0.1.tar.gz
+      #  sha256: a2feb3f1cf002a2e4e8900f718cc5c54db4ad174e48bfcfbddcd588c7b716d5b
+      # - type: dir
+      #   path: ..
+      #   dest: main
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.yml:52]:
```suggestion
```

Leftover?

File: fr.nytuo.cosmiccomics.yml (hunk 3 above)
Comment (Nytuo) [reply fr.nytuo.cosmiccomics.yml:16]: I only use electron as a client for the front end of my application, to give the user a pleasant desktop experience.
The application is mainly intended to be hosted on a server with both a backend and/or a frontend server.
The Electron version run the server as a background child process and the frontend from local files.
So, I don't rely on electron's exclusive features such as the showOpenDialog.

File: fr.nytuo.cosmiccomics.desktop
Comment (Nytuo) [reply fr.nytuo.cosmiccomics.desktop:1]: Ok so I'll have to put it on the app repo (with the metainfo) when it's accepted, am I right?

File: fr.nytuo.cosmiccomics.yml (hunk 3 above)
Comment (hfiguiere) [reply fr.nytuo.cosmiccomics.yml:16]: so what the whole point of giving host access?

File: fr.nytuo.cosmiccomics.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/cosmiccomics/flatpak-node/cache
+        npm_config_cache: /run/build/cosmiccomics/flatpak-node/npm-cache
+        npm_config_offline: "true"
+    build-commands:
+      # Install npm dependencies
+      - npm install --offline
+      - chmod a+x ./node_modules/7zip-bin/linux/x64/7za
```
Comment (hfiguiere) [inline fr.nytuo.cosmiccomics.yml:33]: this will work great on aarch64.

File: fr.nytuo.cosmiccomics.desktop
Comment (hfiguiere) [reply fr.nytuo.cosmiccomics.desktop:1]: you can do it now. these files are not flatpak specific. (and then next release they won't be necessary here)

File: fr.nytuo.cosmiccomics.yml (hunk 3 above)
Comment (Nytuo) [reply fr.nytuo.cosmiccomics.yml:16]: The server must have access to the file system in order to read files from the path specified by the user in the frontend. The fact that the file system is defined as host is necessary in cases where the user has his files stored outside his home directory (such as an external disk mounted in /mnt or /media, for example). Giving the user the right to use only the home directory is of course possible, but the user cannot access files outside this directory.

File: fr.nytuo.cosmiccomics.yml (hunk 6 above)
Comment (Nytuo) [reply fr.nytuo.cosmiccomics.yml]: Thanks, didnt notice it, I've reworked the strategy to avoid chmod and platform specific command

**Comment by barthalion** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by Nytuo** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by Nytuo** [comment]
Hi, I got this error, but I don't know what to do in order to correct it... Do I need to cp the xml to the folder ?

```bash
zgrep "<id>fr.nytuo.cosmiccomics\(\.\w\+\)*\(.desktop\)\?</id>" builddir/*/share/app-info/xmls/fr.nytuo.cosmiccomics.xml.gz
 in dir /srv/buildbot/worker/build-x86_64-5/build (timeout 1200 secs)
 watching logfiles {}
 argv: b'zgrep "<id>fr.nytuo.cosmiccomics\\(\\.\\w\\+\\)*\\(.desktop\\)\\?</id>" builddir/*/share/app-info/xmls/fr.nytuo.cosmiccomics.xml.gz'
 using PTY: False
gzip: builddir/*/share/app-info/xmls/fr.nytuo.cosmiccomics.xml.gz: No such file or directory
program finished with exit code 2
elapsedTime=0.008387```

File: flathub.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+    "only-arches": [
+        "x86_64"
```
Comment (hfiguiere) [inline flathub.json:3]: now that is disappointing. I thought the idea Electron is to be cross-platform.

**Comment by hfiguiere** [comment]
you never install the appstream file, dekstop file, etc. that's why.

File: flathub.json (hunk 7 above)
Comment (Nytuo) [reply flathub.json:3]: Unfortunatly, the server relies on a lib that do not support ARM. So I need to disable support for that arch

**Comment by Nytuo** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by Nytuo** [comment]
bot, build fr.nytuo.cosmiccomics

File: fr.nytuo.cosmiccomics.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  <summary>Comics and manga reader and collectionner</summary>
+  
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+
+  <content_rating type="oars-1.1" />
+
+  <developer id="nytuo">
+    <name>Arnaud BEUX</name>
+  </developer>
```
Comment (bbhtt) [inline fr.nytuo.cosmiccomics.metainfo.xml:15]: Use `developer_name`, Flathub does not yet support developer tag

**Comment by bbhtt** [comment]
Is this primarily meant to run as a server? If so then you'll be limited by flatpak capabilities.

File: fr.nytuo.cosmiccomics.metainfo.xml (hunk 8 above)
Comment (Nytuo) [reply fr.nytuo.cosmiccomics.metainfo.xml:15]: In the doc the developer_name tag is deprecated so I've change it.
i'll revert it to a developer_name tag then

**Comment by Nytuo** [comment]
> Is this primarily meant to run as a server? If so then you'll be limited by flatpak capabilities.

This version is intended to be used locally by the user (like a traditional application). So, yes, under the hood it's a server, but it's only intended to run on localhost. So you can't use it on the local network or over the Internet using Flatpak but that is not a problem since this is not the goal of this version.

**Comment by barthalion** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by barthalion** [comment]
Can you provide a custom first time setup form that will skip questions about username, password and server port altogether? That would improve the onboarding experience a lot.

**Comment by Nytuo** [comment]
Since the app can have multiple accounts and rely on them to work, the username field is mandatory and password is optional but can be set.
However it is true that the server port field is not needed here, so I have removed it.

**Comment by Nytuo** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by barthalion** [comment]
bot, build fr.nytuo.cosmiccomics

**Comment by barthalion** [comment]
Getting an empty window with the latest build. @bbhtt mind testing on your end?

**Comment by bbhtt** [comment]
Seems to be working on my end 

image

**Pull request closed without merging**